modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useResizeObserver.es.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useEffect, useRef, useState} from 'react';
+
+export function useResizeObserver(targetRef) {
```

Comment:
> If ResizeObserver is not supported in some browsers of our matrix we must guarantee a fallback for this, otherwise, it will broken the Actions component.

https://caniuse.com/resizeobserver AFAIK We are supporting this feat 7.4+

Reply:
Presuming this is not going to be backported, seems fine to use `ResizeObserver`, but the `throw` in the `else` is not a good idea. If somebody hits this code path in a non-supported browser we should degrade gracefully (ie. not resize properly, or resize slowly, rather than producing console spew).

Note also that your `if` condition in the callback that you're returning can never be `false` (seeing as you threw an error, it can't get to that point). Alternative that doesn't throw:

```js
useEffect(() => {
    if ('ResizeObserver' in window) {
        const observer = new ResizeObserver(...);

        // ...

        return () => observer.disconnect();
    }
}, [targetRef]);
```

Reply:
Thanks @wincent and @diegonvs